src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_RefReadOnly.cs

```diff
@@ -570,25 +986,883 @@ public class ReadOnlyAttribute : System.Attribute { }
+
+            compilation.VerifyDiagnostics(
+                // (7,20): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsReadOnlyAttribute' is not defined or imported
+                //         void local(ref readonly int x) { }
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound, "ref readonly int x").WithArguments("System.Runtime.CompilerServices.IsReadOnlyAttribute").WithLocation(7, 20));
+        }
+
+        [Fact]
+        public void RefReadOnlyDefinitionsInsideUserDefinedIsReadOnlyAttribute_Class_NoParent()
+        {
+            // PROTOTYPE(readonlyRefs) when the following bug is fixed that test should fail:
+            // https://github.com/dotnet/roslyn/issues/19395
+
+            var code = @"
+namespace System.Runtime.CompilerServices
+{
+    public class IsReadOnlyAttribute
+    {
+        private int value;
+
+        public ref readonly int Method(ref readonly int x) => ref value;
+
+        public static int operator +(ref readonly IsReadOnlyAttribute x, ref readonly IsReadOnlyAttribute y) => 0;
+
+        public ref readonly int Property => ref value;
+
+        public ref readonly int this[ref readonly int x] => ref value;
+    }
+}";
+
+            CompileAndVerify(code, verify: false, symbolValidator: module =>
+            {
+                var isReadOnlyAttributeName = WellKnownTypes.GetMetadataName(WellKnownType.System_Runtime_CompilerServices_IsReadOnlyAttribute);
+                var type = module.ContainingAssembly.GetTypeByMetadataName(isReadOnlyAttributeName);
+
+                var method = type.GetMethod("Method");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, method.GetReturnTypeAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, method.Parameters.Single().GetAttributes(), module.ContainingAssembly.Name);
+
+                var @operator = type.GetMethod("op_Addition");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, @operator.Parameters[0].GetAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, @operator.Parameters[1].GetAttributes(), module.ContainingAssembly.Name);
+
+                var property = type.GetProperty("Property");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, property.GetAttributes(), module.ContainingAssembly.Name);
+
+                var indexer = type.GetProperty("this[]");
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, indexer.GetAttributes(), module.ContainingAssembly.Name);
+                AssertReferencedIsReadOnlyAttribute(Accessibility.Public, indexer.Parameters.Single().GetAttributes(), module.ContainingAssembly.Name);
+            });
+        }
+
+        [Fact]
+        public void RefReadOnlyDefinitionsInsideUserDefinedIsReadOnlyAttribute_Class_CorrectParent()
```

Comment:
>RefReadOnlyDefinitionsInsideUserDefinedIsReadOnlyAttribute_Class_CorrectParent [](start = 20, length = 78)

> Added tests for indexers, operators, properties, and overrides. There is already a test that verifies we produce errors if IsReadOnly was an interface, because it is lacking the constructor.

I don't see any tests added for explicit interface implementations. I believe I asked for them in the previous review pass.